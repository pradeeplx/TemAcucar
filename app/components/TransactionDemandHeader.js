import React, { Component } from 'react'
import { View, TouchableOpacity, Platform } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import TransactionDemandDescription from "./TransactionDemandDescription"
import TransactionDemandTimeAgo from "./TransactionDemandTimeAgo"

export default class TransactionDemandHeader extends Component {
  handleView() {
    const { demand, onViewDemand } = this.props
    onViewDemand(demand)
  }

  render() {
    const { index, currentUser, demand } = this.props
    const { name, user, created_at } = demand
    return(
      <TouchableOpacity onPress={this.handleView.bind(this)} style={{
        padding: 20,
        backgroundColor: (currentUser.id === user.id ? Colors.lightBlue : Colors.lightPink),
        alignItems: 'center',
      }}>
        <UserImage source={{uri: user.image_url}} size={48} />
        <Sentence style={[{
          color: (currentUser.id === user.id ? Colors.blue : Colors.pink),
          fontSize: 12 * fontFactor(),
          marginTop: 10,
        }, (Platform.OS === 'ios' ? { fontFamily: 'Avenir-Black' } : { fontWeight: 'bold' })]}>
          { name }
        </Sentence>
        <View style={{
          flexDirection: 'row',
        }}>
          <Sentence style={{
            color: Colors.black,
            fontSize: 10 * fontFactor(),
          }}>
            { currentUser.id === user.id ? 'VocÃª' : user.first_name } { 'pediu '}
          </Sentence>
          <TimeAgo time={created_at} lowerCase={true} style={{
            color: Colors.black,
            fontSize: 10 * fontFactor(),
          }} />
        </View>
      </TouchableOpacity>
    )
  }
}
