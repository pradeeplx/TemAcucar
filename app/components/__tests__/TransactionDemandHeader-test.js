import 'react-native'
import React from 'react'
import TransactionDemandHeader from '../TransactionDemandHeader'
import renderer from 'react-test-renderer'

const props = {
  index: 0,
  currentUser: { id: 1 },
  demand: {
    user: {
      id: 1,
      image_url: 'http://google.com',
      first_name: 'foo',
    },
    name: 'bar',
    created_at: 10E4,
  },
}

it('renders correctly', () => {
  const tree = renderer.create(
    <TransactionDemandHeader {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders user name and different colors when is not currentUser', () => {
  const tree = renderer.create(
    <TransactionDemandHeader {...props} currentUser={{ id: 2 }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
