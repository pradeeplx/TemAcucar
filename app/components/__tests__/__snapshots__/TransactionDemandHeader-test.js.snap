exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ECF8F7",
      "opacity": 1,
      "padding": 20,
    }
  }
  testID={undefined}>
  <Image
    size={48}
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 36,
          "height": 72,
          "width": 72,
        },
        undefined,
      ]
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Array [
          Object {
            "color": "#44B5B1",
            "fontSize": 18,
            "marginTop": 10,
          },
          Object {
            "fontFamily": "Avenir-Black",
          },
        ],
      ]
    }>
    bar
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#000000",
            "fontSize": 15,
          },
        ]
      }>
      Você
       
      pediu 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "fontSize": 13.5,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ],
        ]
      }>
      há 47 anos
    </Text>
  </View>
</View>
`;

exports[`test renders user name and different colors when is not currentUser 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF2F3",
      "opacity": 1,
      "padding": 20,
    }
  }
  testID={undefined}>
  <Image
    size={48}
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 36,
          "height": 72,
          "width": 72,
        },
        undefined,
      ]
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Array [
          Object {
            "color": "#FF7F85",
            "fontSize": 18,
            "marginTop": 10,
          },
          Object {
            "fontFamily": "Avenir-Black",
          },
        ],
      ]
    }>
    bar
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#000000",
            "fontSize": 15,
          },
        ]
      }>
      foo
       
      pediu 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "fontSize": 13.5,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ],
        ]
      }>
      há 47 anos
    </Text>
  </View>
</View>
`;
